source/agora/consensus/state/ValidatorSet.d

```diff
@@ -736,6 +742,47 @@ public class ValidatorSet
 
         return false;
     }
+
+    /***************************************************************************
+
+        Query Validators that have just finished their cycle
+
+        Params:
+            height = requested height
+
+        Returns:
+            `PublicKey`s and enrollment heights of `Validator`s whose enrollment
+            cycle have just ended
+
+    ***************************************************************************/
+
+    public ExpiringValidator[] getExpiringValidators (Height height,
+        out ExpiringValidator[] ex_validators)
```

Comment:
Needs to be `ref`. `out` sets it to `null`, so setting the `length` later doesn't have the intended effect. Should also call `assumeSafeAppend` after the length is changed to zero.